src/Http/Http.Abstractions/src/Extensions/MapOptions.cs

```diff
@@ -19,5 +19,11 @@ public class MapOptions
         /// The branch taken for a positive match.
         /// </summary>
         public RequestDelegate Branch { get; set; }
+
+        /// <summary>
+        /// If true, matched path would be removed from Request.Path and added to Request.PathBase
+        /// Defaults to true.
+        /// </summary>
+        public bool RemoveMatchedPathSegment { get; set; } = true;
```

Comment:
What would this be called if we had to make it false by default?

Reply:
Only [MapExtensions](https://github.com/aspnet/AspNetCore/blob/425c196cba530b161b120a57af8f1dd513b96f67/src/Http/Http.Abstractions/src/Extensions/MapExtensions.cs#L45) is using `MapOptions`. I intentionally didn't made the default value `false` to prevent a breaking change, in case someone is using `MapMiddleware` directly and without extension method.

Reply:
He's asking to invert the naming, not the behavior.
`KeepMatchedPathSegment`?

Reply:
PreserveMatchedPathSegment

Reply:
Lets invert it.